packages/stats/gbstats/bayesian/tests.py

```diff
@@ -313,3 +323,141 @@ def compute_result(self) -> BayesianTestResult:
+            return self._default_output(ZERO_NEGATIVE_VARIANCE_MESSAGE)
+
+        self.compute_frequentist_moments()
+        self.construct_prior_by_variation()
+
+        mu_a, sd_a = Norm.posterior(
+            [
+                self.prior_a.mean,
+                np.sqrt(self.prior_a.variance),
+                self.prior_a.pseudo_n,
+            ],
+            [
+                self.stat_a.mean,
+                self.stat_a.stddev,
+                self.stat_a.n,
+            ],
+        )
+        mu_b, sd_b = Norm.posterior(
+            [
+                self.prior_b.mean,
+                np.sqrt(self.prior_b.variance),
+                self.prior_b.pseudo_n,
+            ],
+            [
+                self.stat_b.mean,
+                self.stat_b.stddev,
+                self.stat_b.n,
+            ],
+        )
+        self.mean_diff = mu_b - mu_a
+        self.std_diff = np.sqrt(sd_a**2 + sd_b**2)
+        risk = [self.risk]
+        self.var_diff = frequentist_variance(
+            sd_a**2, mu_a, 1, sd_b**2, mu_b, 1, self.relative
+        )
+        self.std_diff = np.sqrt(self.var_diff)
+        self.mean_diff = frequentist_diff(mu_a, mu_b, self.relative)
+        ctw = self.chance_to_win(self.mean_diff, self.std_diff)
+        ci = self.credible_interval(self.mean_diff, self.std_diff, self.ccr, log=False)
+        # probably better to tear these out of superclass, have this be standalone class
+        result = BayesianTestResult(
+            chance_to_win=ctw,
+            expected=self.mean_diff,
+            ci=ci,
+            uplift=Uplift(
+                dist="normal",
+                mean=self.mean_diff,
+                stddev=self.std_diff,
+            ),
+            risk=risk,
+        )
+        if self.scaled:
+            result = self.scale_result(
+                result, self.traffic_proportion_b, self.phase_length_days
+            )
+        return result
+
+    def compute_frequentist_moments(self):
+        self.frequentist_var = frequentist_variance(
+            self.stat_a.variance,
+            self.stat_a.unadjusted_mean,
+            self.stat_a.n,
+            self.stat_b.variance,
+            self.stat_b.unadjusted_mean,
+            self.stat_b.n,
+            self.relative,
+        )
+        self.frequentist_diff = frequentist_diff(
+            self.stat_a.mean, self.stat_b.mean, self.relative
+        )
+
+    def construct_prior_by_variation(self):
+        mu_0 = self.stat_a.mean
+        if self.relative:
+            mu_1 = mu_0 * (1.0 + self.prior_effect.mean)
+            v = (
+                self.prior_effect.variance
+                * mu_0**2
+                / (1.0 + self.prior_effect.mean**2)
+            )
+        else:
+            mu_1 = mu_0 + self.prior_effect.mean
+            v = 0.5 * self.prior_effect.variance
+        self.prior_a = GaussianPrior(mean=mu_0, variance=v, pseudo_n=1)
+        self.prior_b = GaussianPrior(mean=mu_1, variance=v, pseudo_n=1)
```

Comment:
Should the pseudo_n from the `GaussianPrior` be passed through here?

Reply:
I only ask because it seems like there isn't a way to pass through a flat, improper prior to this test class, right? Should we set up a way to do that?

You can kind of hack it into the existing `GaussianPrior` where if `prior_effect.pseudo_n` = 0, then we just pass this through like I suggest. Or we can be more explicit, where you can also specify no prior and that modifies this method as well.

Reply:
good idea, I implemented this.  When we use non-flat priors in the future, we still want to use pseudo-n = 1.